src/app/form/form.component.ts

```diff
@@ -52,19 +52,17 @@ export class FormComponent implements OnInit, OnDestroy {
   }
 
   onSubmit() {
+    let action: UpdateTodo | AddTodo;
     if (this.editTodo) {
-      this.formSubscription.add(
-        this.store.dispatch(new UpdateTodo(this.todoForm.value, this.todoForm.value.id)).subscribe(() => {
-          this.clearForm();
-        })
-      );
+      action = new UpdateTodo(this.todoForm.value, this.todoForm.value.id);
     } else {
-      this.formSubscription.add(
-        this.formSubscription = this.store.dispatch(new AddTodo(this.todoForm.value)).subscribe(() => {
```

Comment:
In this section, it accidentally reinitiates the `formSubscription` and removes the previously added subscriptions (for example, the `selectedTodo` subscription).